corehq/apps/geospatial/static/geospatial/js/case_grouping_map.js

```diff
@@ -158,10 +158,47 @@ hqDefine("geospatial/js/case_grouping_map",[
                 },
             });
         });
+        mapboxInstance.on('moveend', updateClusterStats);
 
         return mapboxInstance;
     }
 
+    function updateClusterStats() {
+        const sourceFeatures = map.querySourceFeatures('caseWithGPS', {
+            filter: ['get', 'cluster'],
```

Comment:
Amazing!

I had a hard time figuring this out, on how to get stats for clusters on the map.
Just curious to how did you find this.

Reply:
I imagine I had the same problem that you experienced, where I struggled finding anything too helpful online. What eventually gave me a good place to start, suprisingly, was ChatGPT. I asked it to generate a function for calculating the total number of clusters and, although the function was far from the final solution, it did give a great starting point and some ideas on how I could approach this.

Reply:
oh wow! I would have never thought of that.